tools/scripts/build.ps1

```diff
@@ -1,14 +1,32 @@
+.\gradlew --stop
 choco install ant
 choco install vswhere
 choco install zip
+if ($env:APPVEYOR -ne "true") {
+  choco install visualstudio2017community
+  choco install visualstudio2017-workload-nativedesktop
+  choco install windows-sdk-7.1
+  choco install cygwin
```

Comment:
I presume that if the user already has installed cygwin via some other manual means, then this will be a no-op?

Reply:
I thought it would be, but it doesn't seem like it. I think I can do a better job of detecting if they're installed before trying to install them.